File: hw11-webflux/src/main/java/ru/otus/hw/models/Comment.java
package ru.otus.hw.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DocumentReference;
import org.springframework.data.relational.core.mapping.Table;

@Getter
@Setter
@RequiredArgsConstructor
@AllArgsConstructor
@Table(name = "comments")
public class Comment {

    @Id
    private Long id;

    @NotNull
    private String text;

    @DocumentReference
    private Book book;
}

File: hw11-webflux/src/main/ui-webapp/src/app/model/comment.ts
import {Book} from "./book";

export class Comment {
  id?: string;
  text?: string;
  book?: Book;

  constructor(id?: string,
              text?: string,
              book?: Book) {
    this.id = id;
    this.text = text;
    this.book = book;
  }
}
